packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/queries.ts

```diff
@@ -0,0 +1,73 @@
+/* Total and circulating supply
+
+REVIEW: https://github.com/input-output-hk/cardano-graphql/blob/610424b95053078c20be121d823ef0dd035711f0
+/packages/api-cardano-db-hasura/src/HasuraClient.ts#L76-L125
+
+*/
+export const findCirculatingSupply = `
+    WITH total_rewards AS (
+        SELECT COALESCE(SUM(amount), 0) as rewards_amount
+        FROM reward
+    ),
+    total_withdrawals AS (
+        SELECT COALESCE(SUM(amount), 0) as withdrawals_amount
+        FROM withdrawal
+    ),
+    total_utxo AS (
```

Comment:
I think this is wrong. This table `ada_pots` looks like a snapshot at the moment the epoch ended. So, it doesn't take into account the already spent UTxOs. At `packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/queries.ts ` you can find `findTotalAda` query that can help you to get this value in real time

Reply:
Thanks for the clarification, here in order to find total `circulatingSupply` and `totalSuply` in lovelace, I used the existing [graphql-implementation](https://github.com/input-output-hk/cardano-graphql/blob/610424b95053078c20be121d823ef0dd035711f0/packages/api-cardano-db-hasura/src/HasuraClient.ts#L76-L125) where I can see the GQL queries but can not check their SQL query representations and what actually is executed at DB level. 
1. Following the GQL implementation above, I assumed [reserves](https://github.com/input-output-hk/cardano-graphql/blob/610424b95053078c20be121d823ef0dd035711f0/packages/api-cardano-db-hasura/src/HasuraClient.ts#L123) from the last epoch is the `total supply` from the network info we need to provide, correct me if I'm wrong?

2. About the [total utxo](https://github.com/input-output-hk/cardano-js-sdk/blob/37b95259e90d4c7ba889a87e5a6ffc01019883da/packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/queries.ts#L16) used in the computation for getting the `circulatingSupply`, I need to find the total ADA ( unspent UTxOS minus already spent ones ) ie. to use [findTotalAda](https://github.com/input-output-hk/cardano-js-sdk/blob/dd8037523275ecd9ea13b695a5fff6515918d8a2/packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/queries.ts#L14-L23) query directly?

3. About the [active stake](https://github.com/input-output-hk/cardano-js-sdk/blob/37b95259e90d4c7ba889a87e5a6ffc01019883da/packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/queries.ts#L55-L63) ( stake snapshot from n - 2 epochs ) the latest DB snapshot doesn't have any data fixtures in `epoch_stake` table, so I can not verify it locally.

4. Could we trust [this inspiration](https://github.com/input-output-hk/cardano-db-sync/issues/599#issuecomment-869142424) for getting accurate value of live stake?

Reply:
> I can see the GQL queries but can not check their SQL query representations and what actually is executed at DB level

https://hasura.io/learn/graphql/hasura-advanced/performance/3-analyze-query-plans/ is available via the Hasura console, or you might get a benefit from checking out the [views](https://github.com/input-output-hk/cardano-graphql/blob/master/packages/api-cardano-db-hasura/hasura/project/migrations/1589369664961_init/up.sql).

> I assumed [reserves](https://github.com/input-output-hk/cardano-graphql/blob/610424b95053078c20be121d823ef0dd035711f0/packages/api-cardano-db-hasura/src/HasuraClient.ts#L123) from the last epoch is the [total supply](https://github.com/input-output-hk/cardano-js-sdk/blob/37b95259e90d4c7ba889a87e5a6ffc01019883da/packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/queries.ts#L24-L32) of the network info we need to provide, correct me if I'm wrong?

`maxSupply - reserves as of last snapshot`
https://github.com/input-output-hk/cardano-graphql/blob/610424b95053078c20be121d823ef0dd035711f0/packages/api-cardano-db-hasura/src/executableSchema.ts#L157-L159

> I need to find the total ADA

Total _coins_ in domain terminology. Check out the [Value](https://github.com/input-output-hk/cardano-js-sdk/blob/606f89ecef27885c2fff4536f9f301554e2ffff0/packages/core/src/Cardano/types/Value.ts#L15) type

> unspent UTxOS minus already spent ones

"Unspent" is redundant, and already spent ones are therefore no longer unspent, so it's technically just the current UTxOs. The database structure make it seem more complicated :-) 

> to use [findTotalAda](https://github.com/input-output-hk/cardano-js-sdk/blob/dd8037523275ecd9ea13b695a5fff6515918d8a2/packages/cardano-services/src/StakePoolSearch/DbSyncStakePoolSearchProvider/queries.ts#L14-L23) query directly, correct

Yes, that would just get the UTxO set, but the economic "circulating supply" includes rewards that have not yet been _withdrawn_, which is represented [here](https://github.com/input-output-hk/cardano-graphql/blob/610424b95053078c20be121d823ef0dd035711f0/packages/api-cardano-db-hasura/src/HasuraClient.ts#L122).

> the latest DB snapshot doesn't have any data fixtures in epoch_stake table, so I can not verify it locally.

The snapshot will need to be updated. Let's chat about this.

> Could we trust https://github.com/input-output-hk/cardano-db-sync/issues/599#issuecomment-869142424 for getting accurate value of [live stake](https://github.com/input-output-hk/cardano-js-sdk/blob/aacf6b429e22eeed6220307f77825980d0334a0f/packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/queries.ts#L44-L58)?

Run with it, for now, then we can analyse the query and cross-reference against other known implementations.